File: android/app/src/main/java/com/redscreenfilter/data/database/UsageEvent.kt
package com.redscreenfilter.data.database

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * UsageEvent Entity
 * Represents a single usage event logged by the analytics system
 */
@Entity(tableName = "usage_events")
data class UsageEvent(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val timestamp: Long,
    val eventType: String, // "overlay_toggled", "settings_changed", "preset_applied"
    val overlayEnabled: Boolean,
    val opacity: Float,
    val preset: String = ""
)

File: ios/RedScreenFilter/Sources/Models/UsageEvent.swift
//
//  UsageEvent.swift
//  RedScreenFilter
//
//  Created on March 5, 2026.
//  Updated for Core Data on March 6, 2026.
//

import Foundation
import CoreData

/// Core Data entity representing a single usage event
@objc(UsageEvent)
public class UsageEvent: NSManagedObject {
    @NSManaged public var timestamp: Date
    @NSManaged public var overlayEnabled: Bool
    @NSManaged public var opacity: Float
    @NSManaged public var preset: String
}

extension UsageEvent {
    @nonobjc public class func fetchRequest() -> NSFetchRequest<UsageEvent> {
        return NSFetchRequest<UsageEvent>(entityName: "UsageEvent")
    }
    
    /// Convenience initializer for creating usage events
    convenience init(context: NSManagedObjectContext, overlayEnabled: Bool, opacity: Float, preset: String) {
        self.init(context: context)
        self.timestamp = Date()
        self.overlayEnabled = overlayEnabled
        self.opacity = opacity
        self.preset = preset
    }
}
